service/history/commandChecker.go

```diff
@@ -191,23 +191,63 @@ func (c *workflowSizeChecker) checkIfNumChildWorkflowsExceedsLimit() error {
 		tag.WorkflowRunID(key.RunID),
 	)
 
-	numPending := len(c.mutableState.GetPendingChildExecutionInfos())
-	errLimit := c.numPendingChildExecutionsLimit
 	if withinLimit(numPending, errLimit) {
 		return nil
 	}
-	c.metricsHandler.Counter(metrics.NumPendingChildWorkflowsTooHigh.GetMetricName()).Record(1)
+	c.metricsHandler.Counter(metricName).Record(1)
 	err := fmt.Errorf(
-		"the number of pending child workflow executions, %d, "+
-			"has reached the error limit of %d established with %q",
+		"the number of %s, %d, has reached the per-workflow limit of %d",
+		resourceName,
 		numPending,
 		errLimit,
-		dynamicconfig.NumPendingChildExecutionsLimitError,
 	)
 	logger.Error(err.Error(), tag.Error(err))
```

Comment:
nit: you can do
```
logger.Error("some error message", 
    tag.WorkflowNamespaceID(key.NamespaceID),
    tag.WorkflowID(key.WorkflowID),
    tag.WorkflowRunID(key.RunID),
    tag.Error(err))
```

Reply:
Yeah, I did it this way because originally I logged a warning as well. I'll clean up in a follow up